lib/interface-types/src/encoders/wat.rs

```diff
@@ -0,0 +1,716 @@
+            InterfaceType::Any => "Any".into(),
+            InterfaceType::String => "String".into(),
+            InterfaceType::Seq => "Seq".into(),
+            InterfaceType::I32 => "i32".into(),
+            InterfaceType::I64 => "i64".into(),
+            InterfaceType::F32 => "f32".into(),
+            InterfaceType::F64 => "f64".into(),
+            InterfaceType::AnyRef => "anyref".into(),
+        }
+    }
+}
+
+/// Encode an `Instruction` into a string.
+impl<'input> From<&Instruction<'input>> for String {
+    fn from(instruction: &Instruction) -> Self {
+        match instruction {
+            Instruction::ArgumentGet { index } => format!("arg.get {}", index),
+            Instruction::Call { function_index } => format!("call {}", function_index),
+            Instruction::CallExport { export_name } => format!(r#"call-export "{}""#, export_name),
+            Instruction::ReadUtf8 => "read-utf8".into(),
+            Instruction::WriteUtf8 { allocator_name } => {
+                format!(r#"write-utf8 "{}""#, allocator_name)
+            }
+            Instruction::AsWasm(interface_type) => {
+                format!("as-wasm {}", String::from(interface_type))
+            }
+            Instruction::AsInterface(interface_type) => {
+                format!("as-interface {}", String::from(interface_type))
+            }
+            Instruction::TableRefAdd => "table-ref-add".into(),
+            Instruction::TableRefGet => "table-ref-get".into(),
+            Instruction::CallMethod(index) => format!("call-method {}", index),
+            Instruction::MakeRecord(interface_type) => {
+                format!("make-record {}", String::from(interface_type))
+            }
+            Instruction::GetField(interface_type, field_index) => {
+                format!("get-field {} {}", String::from(interface_type), field_index)
+            }
+            Instruction::Const(interface_type, value) => {
+                format!("const {} {}", String::from(interface_type), value)
+            }
+            Instruction::FoldSeq(import_index) => format!("fold-seq {}", import_index),
+            Instruction::Add(interface_type) => format!("add {}", String::from(interface_type)),
+            Instruction::MemToSeq(interface_type, memory) => format!(
+                r#"mem-to-seq {} "{}""#,
+                String::from(interface_type),
+                memory
+            ),
+            Instruction::Load(interface_type, memory) => {
+                format!(r#"load {} "{}""#, String::from(interface_type), memory)
+            }
+            Instruction::SeqNew(interface_type) => {
+                format!("seq.new {}", String::from(interface_type))
+            }
+            Instruction::ListPush => "list.push".into(),
+            Instruction::RepeatUntil(condition_index, step_index) => {
+                format!("repeat-until {} {}", condition_index, step_index)
+            }
+        }
+    }
+}
```

Comment:
If the speed is ever a concern here we can implement these `X -> String` functions of a `&mut T` where `T: Write` to avoid the extra allocations.  Very low priority right now (and until the speed becomes a major concern), just writing this for future reference.

Reply:
True. Just wanted a very simple version for debugging purposes first.